src/sink/send.rs

```diff
@@ -32,7 +32,11 @@ impl<S: Sink> Send<S> {
         self.sink.as_mut().take().expect("Attempted to poll Send after completion")
     }
 
-    fn take_sink(&mut self) -> S {
+    /// Consumes this combinator, returning the underlying sink.
+    ///
+    /// Note that this will discard the item being sent if this Future has not completed, so care
+    /// should be taken to avoid losing resources when this is called.
+    pub fn into_inner(&mut self) -> S {
```

Comment:
I'm a little wary about this because like the previous PR this probably wants to return an option, but here it's particularly tricky due to ownership. Could this be left out for now?

Reply:
I just included it for completeness, I don't actually need it because in the shutdown code I write.